odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,197 @@ def __repr__(self):
                                          self.__norm.pwisenorm.exponent)
 
 
+class IndicatorSimplex(Functional):
+
+    r"""Indicator functional of simplex.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \begin{cases}
+            0 & \text{if } x_i \geq 0 \forall i \text{ and } \sum_i x_i = r \\
+            +\infty & \text{else.}
+        \end{cases}
+
+    where `r` is the diameter.
+    """
+
+    def __init__(self, space, diameter=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        diameter : positive float, optional
+            Diameter `r` of simplex.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+        self.diameter = float(diameter)
+
+    def _call(self, x, ndigits=None):
```

Comment:
:+1: for exposing this as a user option. Unfortunately, this operator can't be instantiated since it's not allowed to add arguments to `_call`:
```python
>>> ind = odl.solvers.functional.default_functionals.IndicatorSimplex(odl.rn(3))
...
ValueError: bad signature '_call(self, x, ndigits=None)': output parameter must be called 'out', got 'ndigits'
Possible signatures are ('[, **kwargs]' means optional):

_call(self, x[, **kwargs])
_call(self, x, out[, **kwargs])
_call(self, x, out=None[, **kwargs])
_call(self, x, *, out=None[, **kwargs])

Static or class methods are not allowed.
```
If you really want to add it here, you can use `**kwargs` in the signature and then do `ndigits = kwargs.pop('ndigits', None)`, then proceed as now.

But to me it seems better to add it to the initializer, because that exposes it to users. In `_call`, no-one will find it, and things like `OperatorComp` won't be able to use it.

It might also be easier to convey the meaning of the parameter by using a relative tolerance, e.g. `sum_rtol`, and then comparing the normalized sum to 1: `abs(x.ufuncs.sum() / self.diameter - 1) <= self.sum_rtol`. (`rtol` is a standard name in NumPy).